app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt

```diff
@@ -884,16 +902,21 @@ class ThreadAdapter(
 
     fun undoUserAttendanceClick(message: Message) {
         val indexOfMessage = items.indexOfFirst { it is MessageUi && it.message.uid == message.uid }.takeIf { it >= 0 }
-        indexOfMessage?.let { notifyItemChanged(it, NotifyType.ONLY_REBIND_CALENDAR_ATTENDANCE) }
+        indexOfMessage?.let { notifyItemChanged(it, NotifyType.OnlyRebindCalendarAttendance) }
+    }
+
+    fun messagesCollapseStateChange(isCollapsable: Boolean) {
+        notifyItemRangeChanged(0, itemCount, NotifyType.MessagesCollapseStateChanged(isCollapsable))
     }
 
     // Only public because it's accessed inside of a test file
-    enum class NotifyType {
-        TOGGLE_LIGHT_MODE,
-        RE_RENDER,
-        FAILED_MESSAGE,
-        ONLY_REBIND_CALENDAR_ATTENDANCE,
-        ONLY_REBIND_EMOJI_REACTIONS,
+    sealed class NotifyType {
+        object ToggleLightMode : NotifyType()
+        object ReRender : NotifyType()
+        object FailedMessage : NotifyType()
+        object OnlyRebindCalendarAttendance : NotifyType()
+        object OnlyRebindEmojiReactions : NotifyType()
+        class MessagesCollapseStateChanged(val isCollapsable: Boolean) : NotifyType()
```

Comment:
Do you have a preference between objects vs data objects and class vs data class here?